cardinal-sdk/src/commonTest/kotlin/com/icure/cardinal/sdk/crypto/KeylessPatientUserTest.kt

```diff
@@ -0,0 +1,314 @@
+import com.icure.utils.InternalIcureApi
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+
+@OptIn(InternalIcureApi::class)
+class KeylessPatientUserTest : StringSpec(
+	{
+		val specJob = autoCancelJob()
+
+		lateinit var hcp: DataOwnerDetails
+		lateinit var hcpApi: CardinalSdk
+		lateinit var patientDetails: DataOwnerDetails
+
+		beforeSpec {
+			initializeTestEnvironment()
+
+			hcp = createHcpUser()
+			hcpApi = hcp.api(specJob)
+			patientDetails = createUserFromExistingPatient(
+				hcpApi.patient.createPatient(
+					hcpApi.patient.withEncryptionMetadata(
+						DecryptedPatient(
+							id = defaultCryptoService.strongRandom.randomUUID(),
+							firstName = "John",
+							lastName = "Doe",
+							note = "This note will not be visible to the patient"
+						)
+					)
+				).shouldNotBeNull(),
+				true
+			)
+		}
+
+		suspend fun getPatientApi(): Pair<CardinalSdk, RawDecryptedExchangeData> {
+			val api = patientDetails.api(
+				specJob,
+				object : CryptoStrategies {
+					override suspend fun generateNewKeyForDataOwner(
+						self: DataOwnerWithType,
+						cryptoPrimitives: CryptoService,
+					): CryptoStrategies.KeyGenerationRequestResult {
+						return CryptoStrategies.KeyGenerationRequestResult.Keyless
+					}
+				},
+			)
+			val exchangeData = api.crypto.keylessCreateExchangeDataTo(hcp.dataOwnerId).shouldNotBeNull()
+			return api to exchangeData
+		}
+
+		suspend fun createCalendarItem(patientApi: CardinalSdk, patient: Patient, patientUser: User): DecryptedCalendarItem {
+			val startTime = currentEpochMs() + Math.random().toInt()
+			return patientApi.calendarItem.createCalendarItem(
+				patientApi.calendarItem.withEncryptionMetadata(
+					base = DecryptedCalendarItem(
+						id = defaultCryptoService.strongRandom.randomUUID(),
+						title = "My appointment",
+						startTime = startTime,
+						endTime = startTime + 3600000, // 1 hour later
+						details = "This note will not be visible to the patient"
+					),
+					patient = patient,
+					user = patientUser,
+					secretId = SecretIdUseOption.UseNone,
+					alternateRootDataOwnerReference = EntityReferenceInGroup(hcp.dataOwnerId)
+				)
+			)
+
+		}
+
+		"A keyless API should not allow the creation of encrypted data if no exchange data is provided" {
+			val patientApi = patientDetails.api(
+				specJob,
+				object : CryptoStrategies {
+					override suspend fun generateNewKeyForDataOwner(
+						self: DataOwnerWithType,
+						cryptoPrimitives: CryptoService,
+					): CryptoStrategies.KeyGenerationRequestResult {
+						return CryptoStrategies.KeyGenerationRequestResult.Keyless
+					}
+				},
+			)
+
+			val patientUser = patientApi.user.getCurrentUser()
+			val patient = patientApi.patient.encrypted.getPatient(patientUser.dataOwnerId).shouldNotBeNull()
+
+			runCatching {
+				createCalendarItem(patientApi, patient, patientUser)
+			}.isFailure shouldBe true
```

Comment:
use `shouldThrow<SomeException>` instead. 
Need to decide which type of exception is appropriate (illegal state?) and also verify that